portal-web/test/functional/com/liferay/portalweb/tests/Homepage.testcase

```diff
@@ -0,0 +1,37 @@
+definition {
+
+	property testray.main.component.name = "Login";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "Test Test");
+		}
+	}
+
+	@description = "LRQA-82341 Post Login Admin can view Welcome to Liferay text"
+	test VerfiyUserName {
+		Click.waitForPersonalMenuJSClick(locator1 = "Homepage#USER_PROFILE_MENU");
+
+		AssertTextEquals(
+			locator1 = "Homepage#USER_NAME",
+			value1 = "Test Test");
+
+		AssertTextEquals(
+			locator1 = "Homepage#PAGE_HEADING",
+			value1 = "Welcome to Liferay");
```

Comment:
@Piyush-Gupta10 while it will certainly work to add functions directly in the testcase file, they should be done very sparingly, and have a good reason to be done that way.  A better approach here would be to have two (or even three) macro commands: one for navigating to USER_PROFILE_MENU, and the second to assert User and Welcome message.  This way, anyone else trying to assert these things won't be writing new function code directly in their testcase file, but they can use the same macro code that you've written.  This way, also, if the locator changes or the message breaks, we can quickly fix the issue because we only have the function declared in this way once - so a fix there is a fix for all tests referencing it.

I can explain this more in our next meeting if it's not totally clear.  Good job making a test work in your first try though :)